WordPress/gutenberg
# Strip HTML from gallery image alt text

Fixes https://github.com/WordPress/gutenberg/issues/26549

## Description
When a gallery image's `alt` attribute value is empty, we use what's in the `caption` prop to fill it out. The issue is that `caption` can contain links, whereas the `alt` attribute should not contain any HTML. This PR makes sure the `alt` attribute when created from `caption`, receives a value stripped out of HTML.

## How has this been tested?
Manually:
1. Add the Image Gallery block.
2. Add some images.
3. Fill the caption with some text and a link.
4. Preview the site and inspect the image element.
5. See that the `alt` attribute doesn't contain any HTML.

To run unit tests covering the change:

```
npm run test-unit -- test/integration/full-content/full-content.test.js
```

## Types of changes
Bugfix

## Checklist:

## TODO
- [ ] Figure out a way to access 'implicit' attributes (that aren't stored in the comment, but inferred through a query) in `migrate`
- [ ] Migration: Unescape HTML in `alt` attribute.
